training_main.py

```diff
@@ -0,0 +1,354 @@
+
+#%%
+
+# Importing custom modules
+
+from utils import image_processing
+from utils import caption_writer
+from training_modules import description_processor
+from training_modules import description_properties
+from training_modules import training_functions
+
+# %%
+
+# Loading training descriptions into 'descriptions' dictionary
+
+descriptions= description_processor.load_descriptions()
+
+# %%
+
+# Printing Sample-keys and Sample-descriptions
+
+print("Sample keys : ", list(descriptions.keys())[:5], "\n")
+print("Sample Description 1 : ", descriptions['1000268201'], "\n")
+print("Sample Description 2 : ", descriptions['1000344755'], "\n")
+
+# %%
+
+# Cleaning the description strings making it compatible for training model
+
+description_processor.clean_descriptions(descriptions)
+
+# %%
+
+#Printing cleaned description samples
+
+print("Sample 1 : ", descriptions['1000344755'], "\n")
+print("Sample 2 : ", descriptions['1000268201'], "\n")
+    
+# %%
+
+# Adding all distinct words in the description dictionary to the 'all_desc' set
+
+all_desc=set()
+for key in descriptions.keys():
+    #print(key)
+    for d in descriptions[key]:
+        [all_desc.update(d.split())]
+
+# %%
+
+# Printing the number of distinct words present in the descriptions
+        
+vocabulary=all_desc
+print("size of the vocabulary=",len(vocabulary))
+
+# %%
+
+# Loading and printing the size of the description dataset
+
+filename='results.csv'
+train=description_processor.load_set(filename)
+print('dataset=',len(train))
+
+# %%
+
+# Initializing the images-dataset directory and adding the name of each image into 'img' list
+
+images='./flickr30k_images/flickr30k_images/'
+img=glob.glob(images+'*.jpg')
+
+# %%
+
+# Initializing and customizing the InceptionV3 model
+
+model=InceptionV3(weights='imagenet')
+model_new=Model(model.input,model.layers[-2].output)
+
+# %%
+
+# Setting time-stamp as current time and initializing empty dictionary 'encoding_train'
+start=time()
+encoding_train={}
+
+# Looping through and encoding all the images in the train directory
+
+for img in img:
+    encoding_train[img[len(images):]]=image_processing.encode(img)
+    print("Time taken in second=",time()-start)
+
+# %%
+
+# Pickling the encoded training images dataset 
+    
+with open('./resources/encoded_train_images.pkl','wb') as f:
+    dump(encoding_train,f)
+
+# %%
+
+# Loading the pickled encoded-image dataset and initializing it as the training feature matrix
+    
+train_features=load(open('./resources/encoded_train_images.pkl','rb'))
+print(len(train_features))
+
+# %%
+
+# Adding all the training captions to a list and printing it's length
+
+all_train_captions=[]
+for key,val in descriptions.items():
+    for cap in val:
+        all_train_captions.append(cap)
+len(all_train_captions)
+
+# %%
+
+# Prints the word count of the description dataset and initializes the frequent-words list
+
+vocab=description_properties.description_vocabulary(all_train_captions)
+
+# %%
+
```

Comment:
dictionaries should also be created in other files . These functions are just utilities which are helping the model to train